sdk/keyvault/azure-security-keyvault-secrets/src/private/secret_serializers.hpp

```diff
@@ -111,14 +111,14 @@ namespace Azure { namespace Security { namespace KeyVault { namespace Secrets {
     static std::string Serialize(std::vector<uint8_t> const& backup);
   };
 
-  class SecretPropertiesPagedResultSerializer final {
-  public:
+  struct SecretPropertiesPagedResultSerializer final
```

Comment:
well, I am fighting here: https://github.com/Azure/azure-sdk-for-cpp/pull/2763#discussion_r691466229 to use `struct` for these type of components which exposes only public methods.

But, I am loosing the fight after the last comment about these methods should be deleting its constructor (or keep it private) to avoid people creating instances of this class.  So, you might want to keep the `class` and add the `public` keywork instead.

Reply:
the rest of them are structs , as they only have static methods, changed it to struct for consistency, i missed it here and for soem reason made it a class , even though it has only one public method. think of it a syntactical sugar , also these classes are in in private folder, which is not published, thus the h files cannot be included by others.